dali/pipeline/operators/color/brightness_contrast.h

```diff
@@ -0,0 +1,197 @@
+  GetSingleOrRepeatedArg(spec, arg, arg_name);
+}
+
+}  // namespace detail
+
+
+template <typename Backend>
+class BrightnessContrast : public Operator<Backend> {
+ public:
+  explicit BrightnessContrast(const OpSpec &spec) :
+          Operator<Backend>(spec),
+          output_type_(spec.GetArgument<DALIDataType>(spec::kOutputType)) {
+    detail::assign_argument_value<Backend>(spec, spec::kBrightness, brightness_);
+    detail::assign_argument_value<Backend>(spec, spec::kContrast, contrast_);
+    if (std::is_same<Backend, GPUBackend>::value) {
+      kernel_manager_.Resize(1, 1);
+    } else {
+      kernel_manager_.Resize(num_threads_, batch_size_);
+    }
+  }
+
+
+  ~BrightnessContrast() = default;
+
+
+  DISABLE_COPY_MOVE_ASSIGN(BrightnessContrast);
+
+
+ protected:
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+
+  bool SetupImpl(std::vector<::dali::OutputDesc> &output_desc,
+                 const ::dali::workspace_t<Backend> &ws) override {
+    const auto &input = ws.template InputRef<Backend>(0);
+    const auto &output = ws.template OutputRef<Backend>(0);
+    output_desc.resize(1);
+
+    TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float), (
+        {
+          TypeInfo type;
+          type.SetType<OutputType>(output_type_);
+          output_desc[0] = {input.shape(), type};
+        }
+    ), DALI_FAIL("Unsupported image type"))  // NOLINT
+    return true;
+  }
+
+  /*
+   * So that compiler wouldn't complain, that
+   * "overloaded virtual function `dali::Operator<dali::CPUBackend>::RunImpl` is only partially
+   * overridden in class `dali::brightness_contrast::BrightnessContrast<dali::CPUBackend>`"
+   */
+  using Operator<Backend>::RunImpl;
+
+
+  void RunImpl(Workspace<Backend> &ws) {
+    const auto &input = ws.template Input<Backend>(0);
+    auto &output = ws.template Output<Backend>(0);
+
+    TYPE_SWITCH(input.type().id(), type2id, InputType, (uint8_t, int16_t, int32_t, float), (
+        TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float), (
+        {
+            using BrightnessContrastKernel =
+                    detail::BrightnessContrastKernel<Backend, OutputType, InputType, 3>;
+            auto tvin = view<const InputType, 3>(input);
+            auto tvout = view<OutputType, 3>(output);
+            kernel_manager_.Initialize<BrightnessContrastKernel>();
+            kernels::KernelContext ctx;
+            kernel_manager_.Setup<BrightnessContrastKernel>(ws.data_idx(),
+                    ctx, tvin, brightness_, contrast_);
+            kernel_manager_.Run<BrightnessContrastKernel>(ws.thread_idx(), ws.data_idx(),
+                    ctx, tvout, tvin, brightness_, contrast_);
+        }
+        ), DALI_FAIL("Unsupported output type"))  // NOLINT
+    ), DALI_FAIL("Unsupported input type"))  // NOLINT
+  }
+
+
+ private:
+  USE_OPERATOR_MEMBERS();
+  detail::argument_t<Backend> brightness_, contrast_;
```

Comment:
Maybe personal preference:
A little bit of an overkill in my opinion. I'd rather have `vector<float>` all the time and get rid of `argument_t`. It will just have 1 element for CPUBackend and more than that for GPUBackend